### PR TITLE
feat: expose animation function to customize animation

### LINKED ISSUE #8: Feature Request: An option to remove transition animation
Hi,

I was reviewing your latest changes and it's going good. Can you consider adding an option to disable pop up animation for bottom-sheet?

### PR DESCRIPTION
**Pull request by hasanparasteh**
Closes: #8 

I created a feature to manually pass a function to sheet component for handling enter & leave animation! if there was no prop it uses the default animation behavior so it's fully compatible with previous version.

I'd be happy if you review my codes and merge this PR

### CODE REVIEW

File: src/Component.tsx [hunk 1]
```diff
@@ -304,8 +304,17 @@ export const Sheet = defineComponent({
...
+            props.handleAnimationEnter !== undefined ?
+              props.handleAnimationEnter :
+              handleAnimationEnter 
+              
+          }
+          onLeave={
+            props.handleAnimationLeave !== undefined ?
+              props.handleAnimationLeave :
+              handleAnimationLeave
+          }
```
Comment (kadiryazici) [inline src/Component.tsx]: Firstly thanks for the PR, We should treat null and undefined differently for this behavior.

If user passes null: no transition
If user passes undefined: use default transition callback
If user passes a function: use that function as a transition callback.

And I prefer ternary operators in new line :)

```suggestion
          onEnter={
            props.handleAnimationEnter === null
              ? undefined
               : props.handleAnimationEnter || handleAnimationEnter
          }
          onLeave={
            props.handleAnimationLeave === null
              ? undefined :
              props.handleAnimationLeave || handleAnimationLeave
          }
```

File: src/constants.ts [hunk 2]
```diff
@@ -7,6 +7,11 @@ const BOOL_FALSE = {
   default: false,
 } as const
 
+const ANIMATION_CALLBACK = {
+  type: Function as PropType<(backdrop: Element, done: () => void) => void>,
+  default: undefined
+} as const
```
Comment (kadiryazici) [inline src/constants.ts]:
```suggestion
const ANIMATION_CALLBACK = {
  type: Function as PropType<((element: Element, done: () => void) => void) | null | undefined>,
  default: undefined
} as const
```

**Review by kadiryazici — CHANGES_REQUESTED** [review]

**Comment by kadiryazici** [comment]
And prop names should be `animationLeaveHandler` and `animationEnterHandler` because it looks like a function name.

File: src/constants.ts [hunk 3]
```diff
@@ -21,4 +26,7 @@ export const SHEET_PROPS = {
   noClickOutside: BOOL_FALSE,
   /** Removes header section, ignores #header slot */
   noHeader: BOOL_FALSE,
+  /** Loads custom animations callback */
+  handleAnimationEnter: ANIMATION_CALLBACK,
+  handleAnimationLeave: ANIMATION_CALLBACK
```
Comment (kadiryazici) [inline src/constants.ts]: We should allow users to write it like `@transitionEnter` and `@transitionLeave`, vue do not care about if it is prop or an emit if it is prefixed as `on`.

```suggestion
  onTransitionEnter: TRANSITION_CALLBACK,
  onTransitionEnter: TRANSITION_CALLBACK
```

**Review by kadiryazici — CHANGES_REQUESTED** [review]

File: src/Component.tsx [hunk 4]
```diff
@@ -304,8 +304,16 @@ export const Sheet = defineComponent({
...
+          onEnter={
+            props.handleAnimationEnter === null
+              ? undefined
+               : props.handleAnimationEnter || handleAnimationEnter
+          }
+          onLeave={
+            props.handleAnimationLeave === null
+              ? undefined :
+              props.handleAnimationLeave || handleAnimationLeave
+          }
```
Comment (kadiryazici) [inline src/Component.tsx]:
```suggestion
            props.handleAnimationEnter === null
              ? undefined
              : props.handleAnimationEnter || handleAnimationEnter
          }
          onLeave={
            props.handleAnimationLeave === null
              ? undefined
              : props.handleAnimationLeave || handleAnimationLeave
          }
```

**Review by kadiryazici — CHANGES_REQUESTED** [review]

File: src/Component.tsx [hunk 5]
```diff
@@ -304,8 +304,16 @@ export const Sheet = defineComponent({
...
+          onEnter={
+            props.handleAnimationEnter === null
+              ? undefined
+              : props.handleAnimationEnter || handleAnimationEnter
+          }
+          onLeave={
+            props.handleAnimationLeave === null
+              ? undefined
+              : props.handleAnimationLeave || handleAnimationLeave
+          }
```
Comment (kadiryazici) [inline src/Component.tsx:314]: you shouldn't commit my suggestions directly :) these are still staying as props.handleAnimation(start/leave)

**Comment by kadiryazici** [comment]
I'm wondering why `type-check` workflow doesn't run 🤔

**Comment by kadiryazici** [comment]
@hasanparasteh can you pull the latest changes? I updated workflows.

**Comment by hasanparasteh** [comment]
I'll close this PR and try to remove additional and extra commits to make a clean PR.

**Comment by kadiryazici** [comment]
@hasanparasteh No need to create another PR, you just can update this one.

**Pull request closed without merging**